### ISSUE TITLE
Improve config architecture

### ISSUE THREAD

**Issue by thorncp** [thoughtbot/velveteen#4]
Having a convention for a config file could reduce the burden of manually setting config values in different contexts.

Also, memoizing at the module function level has already caused some issues in the test suite. Let's move away from that.

**Comment by thorncp** [comment]
Playing devil's advocate, the current implementation of the config makes it simple to specify whatever is needed. E.g. being able to instantiate the Bunny connection yourself and tell Velveteen about it means Velveteen doesn't have to know those details.

**Comment by thorncp** [comment]
The shared global config struck again with BunnyMock caching queues by name, which caused some data to leak out of tests (#11).

**Comment by eebs** [comment]
We hit this too, but with Bunny itself. 😁 We had some code that cleared the internal Bunny cache when we created the retry queue.

```ruby
def create_retry_queue!(delay)
  clear_queues_cache!

  broker.queue #...
end

# When we create a new queue, `Bunny` stores its name in an internal
# cache. The problem is that as we are creating ephemeral queues that can
# expire shortly after they are created, this cached queue may not exist
# anymore when we try to publish a second message to it.
#
# Removing queues from the cache guarantees that `Bunny` will always try
# to check if they exist, and when they don't, it will create them for us.
def clear_queues_cache!
  broker.with_channel do |channel|
    channel.queues.clear
  end
end
```



I'm not sure if this is relevant here but I figured I'd chime in.

**Comment by thorncp** [comment]
Yep, I ran into that as well! Bunny and BunnyMock have a slightly different interface, i.e. BunnyMock doesn't provide `channel.queues`, which caused some pain when testing. I worked around that by creating the retry queues manually (https://github.com/thoughtbot/velveteen/blob/master/lib/velveteen/error_handlers/exponential_backoff.rb#L62), which bypasses the cache. That approach does contribute to the config awkwardness though 😬 Perhaps a better solution would be to add `channel.queues` upstream to make the two consistent.
